java/java-common/src/main/java/org/cloudburstmc/protocol/java/Java.java

```diff
@@ -0,0 +1,57 @@
+package org.cloudburstmc.protocol.java;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.nukkitx.protocol.MinecraftInterface;
+import io.netty.bootstrap.Bootstrap;
+import io.netty.channel.EventLoopGroup;
+import lombok.Getter;
+import lombok.Setter;
+import net.kyori.adventure.text.serializer.gson.GsonComponentSerializer;
+import org.cloudburstmc.protocol.java.data.profile.GameProfile;
+import org.cloudburstmc.protocol.java.data.profile.property.PropertyMap;
+import org.cloudburstmc.protocol.java.util.json.GameProfileSerializer;
+import org.cloudburstmc.protocol.java.util.json.PropertyMapSerializer;
+import org.cloudburstmc.protocol.java.util.json.UUIDTypeAdapter;
+
+import java.net.InetSocketAddress;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+@Getter
+@Setter
+public abstract class Java implements MinecraftInterface {
+    public static final Gson GSON = GsonComponentSerializer.gson()
+            .populator()
+            .apply(new GsonBuilder()
+                    .registerTypeAdapter(GameProfile.class, new GameProfileSerializer())
+                    .registerTypeAdapter(PropertyMap.class, new PropertyMapSerializer())
+                    .registerTypeAdapter(UUID.class, new UUIDTypeAdapter())
+            )
+            .setPrettyPrinting()
+            .create();
+
+    protected final InetSocketAddress bindAddress;
+    final EventLoopGroup eventLoopGroup;
+    private final Bootstrap bootstrap;
+    protected boolean handleLogin = true;
+
+    Java(InetSocketAddress bindAddress, EventLoopGroup eventLoopGroup) {
```

Comment:
As we're looking into "direct injection" support on some server softwares, this (and other mentions of it) should not be INetSocketAddress but SocketAddress.